File: backend/src/api/feed.py
import falcon
from api.article import Article, ArticleSchema
from api.dataset import Dataset, DatasetSchema
from api.event   import Event, EventSchema
from api.model   import Model, ModelSchema
from api.paper   import Paper, PaperSchema
from api.project import Project, ProjectSchema

class FeedResource:
    def on_get(self, req, resp):
        articles, _ = ArticleSchema(many=True).dump(self.db.query(Article).all())
        datasets, _ = DatasetSchema(many=True).dump(self.db.query(Dataset).all())
        events, _ = EventSchema(many=True).dump(self.db.query(Event).all())
        models, _ = ModelSchema(many=True).dump(self.db.query(Model).all())
        papers, _ = PaperSchema(many=True).dump(self.db.query(Paper).all())
        projects, _ = ProjectSchema(many=True).dump(self.db.query(Project).all())

        [article.update({'type':'article'}) for article in articles]
        [dataset.update({'type':'dataset'}) for dataset in datasets]
        [event.update({'type':'event'}) for event in events]
        [model.update({'type':'model'}) for model in models]
        [paper.update({'type':'paper'}) for paper in papers]
        [project.update({'type':'project'}) for project in projects]

        items = articles + datasets + events + models + papers + projects

        resp.media = sorted(items, key=lambda x: x['created_at']) 

File: koa/src/contexts/feed.js
import mongoose from 'mongoose'
import article from './article'
import dataset from './dataset'
import paper from './paper'
import project from './project'
import video from './video'

const feed = {
  index: async (ctx) => {
    const articles = await article.model.find().lean()
    const datasets = await dataset.model.find().lean()
    const papers = await paper.model.find().lean()
    const projects = await project.model.find().lean()
    const videos = await video.model.find().lean()

    articles.map(article => article.type = 'article')
    datasets.map(dataset => dataset.type = 'dataset')
    papers.map(paper => paper.type = 'paper')
    projects.map(project => project.type = 'project')
    videos.map(video => video.type = 'video')

    const items = [...articles, ...datasets,, ...papers, ...projects, ...videos]
    const sorted = items.sort((a, b) => {return new Date(b.createdAt) - new Date(a.createdAt)})

    ctx.response.body = sorted 
  }
}

export default feed